openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -401,21 +403,29 @@ const GlossaryTermTab = ({ isGlossary, className }: GlossaryTermTabProps) => {
           );
 
           return (
-            <div>
-              {status === Status.InReview && permission ? (
-                <StatusAction
-                  dataTestId={record.name}
-                  onApprove={() => handleApproveGlossaryTerm(taskId, termFQN)}
-                  onReject={() => handleRejectGlossaryTerm(taskId, termFQN)}
-                />
-              ) : (
-                <StatusBadge
-                  dataTestId={termFQN + '-status'}
-                  label={status}
-                  status={StatusClass[status]}
-                />
-              )}
-            </div>
+            <Popover
+              content={
+                <WorkflowHistory glossaryTerm={record as GlossaryTerm} />
+              }
+              overlayStyle={{ maxWidth: '400px' }}
+              placement="topLeft"
+              trigger="hover">
+              <div>
+                {status === Status.InReview && permission ? (
+                  <StatusAction
+                    dataTestId={record.name}
+                    onApprove={() => handleApproveGlossaryTerm(taskId, termFQN)}
+                    onReject={() => handleRejectGlossaryTerm(taskId, termFQN)}
+                  />
+                ) : (
+                  <StatusBadge
+                    dataTestId={termFQN + '-status'}
+                    label={status}
+                    status={StatusClass[status]}
+                  />
+                )}
+              </div>
+            </Popover>
```

Comment:
Need popover only on StatusBadge and not on StatusAction.
Also don't need in case of 'Approved' state